packages/backend/src/income/__tests__/service/getIncomeService.test.ts

```diff
@@ -0,0 +1,80 @@
+import { getIncomeService } from '../../service/getIncomeService'
+import { getIncomeRepository } from '../../repository/getIncomeRepository'
+import { GetIncomeInput } from '../../validation/models'
+
+jest.mock('../../repository/getIncomeRepository', () => ({
+  getIncomeRepository: jest.fn(),
+}))
+
+describe('getIncomeService', () => {
+  const mockedGetIncomeRepository = getIncomeRepository as jest.Mock
+  const fakeInput = {
+    userId: 'user-1',
+    accountId: 'account-1',
+  } as GetIncomeInput
+
+  beforeEach(() => {
+    mockedGetIncomeRepository.mockReset()
+  })
+
+  describe('when repository throws an error', () => {
+    it('should throw an error', async () => {
+      // Arrange
+      const error = new Error('Repository error')
+      mockedGetIncomeRepository.mockRejectedValueOnce(error)
+
+      // Act & Assert
+      await expect(getIncomeService(fakeInput)).rejects.toThrow('Repository error')
+    })
+  })
+
+  describe('when repository returns data', () => {
+    it('should return the data returned from the repository', async () => {
+      // Arrange
+      const fakeIncomeList = [
+        {
+          id: 'income-1',
+          userId: 'user-1',
+          accountId: 'account-1',
+          name: 'Salary',
+          amount: 5000,
+          date: '2023-12-01',
+          createdAt: new Date('2023-12-01'),
+          updatedAt: new Date('2023-12-01'),
+        },
+        {
+          id: 'income-2',
+          userId: 'user-1',
+          accountId: 'account-1',
+          name: 'Bonus',
+          amount: 1000,
+          date: '2023-11-01',
+          createdAt: new Date('2023-11-01'),
+          updatedAt: new Date('2023-11-01'),
```

Comment:
One item is fine in the array and just the id value is fine as well. These are mocked responses so exact object is not needed